library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java

```diff
@@ -205,20 +216,72 @@ void resolveSettingsApi(Status status) {
         }
     }
 
-    boolean checkLastKnowLocation() {
-        if (getSourceProvider().getLocationAvailability()) {
-            Location lastKnownLocation = getSourceProvider().getLastLocation();
-            if (lastKnownLocation != null) {
-                LogUtils.logI("LastKnowLocation is available.");
-                onLocationChanged(lastKnownLocation);
-                return true;
-            } else {
-                LogUtils.logI("LastKnowLocation is not available.");
-            }
+    void checkLastKnowLocation() {
+        getSourceProvider().getLocationAvailability()
+                .addOnSuccessListener(new OnSuccessListener<LocationAvailability>() {
+                    /**
+                     * Returns the availability of location data. When isLocationAvailable() returns true, then the location returned by getLastLocation() will be reasonably up to date within the hints specified by the active LocationRequests.
+                     *
+                     * If the client isn't connected to Google Play services and the request times out, null is returned.
+                     *
+                     * Note it's always possible for getLastLocation() to return null even when this method returns true (e.g. location settings were disabled between calls).
```

Comment:
If so, does it even worth to call `availability` in the first place? It seems like calling `getLastLocation` directly would be sufficient, considering the fact that on its absence it will trigger location request anyway.

Reply:
Delete getLocationAvailability() call, use getLastLocation() directly